### ISSUE TITLE
Issues with Product Export Automation in n8n (OFN Canada)

### ISSUE THREAD

**Issue by div-yansh-1** [openfoodfoundation/integrations#2]
### Description:

Product export automation in n8n is having some issues. Ideally it should give a csv file with all the products/variants for an enterprise in the format required for OFN product upload file. This will allow hubs to edit the CSV file with new stock level and upload the same directly on OFN website.

**Issues mentioned in the [slack thread](https://openfoodnetwork.slack.com/archives/C04U02X3806/p1678991747931379):**

Issue 1 -  the export file is NOT capturing all the products in the enterprise product list.  Products that are missing all have one thing in common.  On the product list,  they are showing an import date.  All products, on my product list, showing an import date,  are not included in the export file.  All products, on my product list, without an import date,  are included in the export file.  (SO - maybe previously imported products are not included in the current n8n download?)

Issue 2 - for this recent export David used a unique login for an enterprise called Garden Party (162) (thats the enterprise I want the export for) But SOME products (and not all) from 2 other enterprises were included in the export file also. The export for this unique login included products from Garden Party Plants (333) and Garden Party Flower Farm (243).   Those 2 enterprise (333 & 243) have given enterprise 162 permission for 'everything',  so perhaps this is why some of their products got included?
I've tried to figure out why some of these 'other enterprise' products are being included in the export file,  but others are not and i can't see anything that defines them.
But for now -  it seems that when doing the n8n export file for a given enterprise,    that enterprise needs to have 1. a unique login,  and ALSO 2. no other enterprise has given 'everything' permissions.

**Comment by tschumilas** [comment]
Re:  it seems that when doing the n8n export file for a given enterprise, that enterprise needs to have 1. a unique login, and ALSO 2. no other enterprise has given 'everything' permissions.
I just removed permissions from the enterprise I was testing -  and indeed,  the n8n export no longer includes the products from those permission-giving enterprises.   So - at present -  as long as the enterprise has a unique login,   and no other enterprise has given it 'manage products' permission,  the 8n8 export pulls the correct products.


Re: (SO - maybe previously imported products are not included in the current n8n download?)
Did you update this since I posted on slack @div-yansh-1 ?   Because I just tested another export file -   same enterprise as 2 days ago -  and now,  the products that were missing in the first test,  are present.  So -  confirming that the n8n export is drawing all the products from the enterprise's product list (no idea why before previously imported products weren't included.)

So it seems like the 8n8 download is ALMOST there.  BUT -   I'm getting an error that 'shipping category cannot be blank' when I go to import the data from the n8n export.   So -  I guess we need to include shipping category data in the export?    If I just put 'default' manually into my import file -  it works (I can get rid of that import error)

AND -  one last thing.   The n8n export needs to include the 'on demand' field.    Right now it seems that if a producer has set a product to 'on demand' in their product list,   the n8n export reads 'on hand' =0.   And then importing using this file,  changes on demand to on hand = 0 in the product list.    The import file template needs an on-demand column,  and the on demand field = 1  when  on demand is set.   Otherwise,  the import file just re-sets on-demand to  on-hand =0.

I am SO SO excited that we are close......
